lib/spack/spack/modules.py

```diff
@@ -674,21 +651,22 @@ class LmodModule(EnvModule):
     path_part_with_hash = join_path('{token.name}', '{token.version}-{token.hash}')  # NOQA: ignore=E501
     path_part_without_hash = join_path('{token.name}', '{token.version}')
 
-    # TODO : Check that extra tokens specified in configuration file
-    # TODO : are actually virtual dependencies
-    configuration = CONFIGURATION.get('lmod', {})
-    hierarchy_tokens = configuration.get('hierarchical_scheme', [])
-    hierarchy_tokens = hierarchy_tokens + ['mpi', 'compiler']
-
     def __init__(self, spec=None):
         super(LmodModule, self).__init__(spec)
+
+        self.configuration = CONFIGURATION.get('lmod', {})
+        hierarchy_tokens = self.configuration.get('hierarchical_scheme', [])
+        # TODO : Check that the extra hierarchy tokens specified in the
+        # TODO : configuration file are actually virtual dependencies
+        self.hierarchy_tokens = hierarchy_tokens + ['mpi', 'compiler']
```

Comment:
Thanks for fixing this.: `hierarchy_tokens` was meant to be set during init + this will solve weird issues when mocking up values in tests.